tests/test_models.py

```diff
@@ -0,0 +1,162 @@
+import json
+import os
+import unittest
+import sys
+
+from common import replace_text_in_file, collect_ap
+
+
+class PublicModelsTestCase(unittest.TestCase):
+    coco_dir = '/tmp/coco'
+    snapshots_dir = '/tmp/snapshots'
+
+    @staticmethod
+    def shorten_annotation(path, num_images):
+        with open(path) as read_file:
+            content = json.load(read_file)
+            selected_indexes = sorted([item['id'] for item in content['images']])
+            selected_indexes = selected_indexes[:num_images]
+            content['images'] = [item for item in content['images'] if
+                                 item['id'] in selected_indexes]
+            content['annotations'] = [item for item in content['annotations'] if
+                                      item['image_id'] in selected_indexes]
+            content['licenses'] = [item for item in content['licenses'] if
+                                   item['id'] in selected_indexes]
+
+        with open(path, 'w') as write_file:
+            json.dump(content, write_file)
+
+    @classmethod
+    def setUpClass(cls):
+        os.makedirs(cls.coco_dir, exist_ok=True)
+        if not os.path.exists(os.path.join(cls.coco_dir, 'val2017.zip')):
+            os.system(f'wget http://images.cocodataset.org/zips/val2017.zip -P {cls.coco_dir}')
```

Comment:
Please add '--no-verbose' to disable progress bar.

Reply:
added